src/librustdoc/html/static/main.js

```diff
@@ -163,59 +163,71 @@ function getSearchElement() {
 
     var main = document.getElementById("main");
 
-    function highlightSourceLines(ev) {
-        // If we're in mobile mode, we should add the sidebar in any case.
+    function onHashChange(ev) {
+        // If we're in mobile mode, we should hide the sidebar in any case.
         hideSidebar();
-        var elem;
-        var search = getSearchElement();
-        var i, from, to, match = window.location.hash.match(/^#?(\d+)(?:-(\d+))?$/);
+        var match = window.location.hash.match(/^#?(\d+)(?:-(\d+))?$/);
         if (match) {
-            from = parseInt(match[1], 10);
-            to = from;
-            if (typeof match[2] !== "undefined") {
-                to = parseInt(match[2], 10);
-            }
-            if (to < from) {
-                var tmp = to;
-                to = from;
-                from = tmp;
-            }
-            elem = document.getElementById(from);
-            if (!elem) {
-                return;
-            }
-            if (ev === null) {
-                var x = document.getElementById(from);
-                if (x) {
-                    x.scrollIntoView();
-                }
-            }
-            onEachLazy(document.getElementsByClassName("line-numbers"), function(e) {
-                onEachLazy(e.getElementsByTagName("span"), function(i_e) {
-                    removeClass(i_e, "line-highlighted");
-                });
-            });
-            for (i = from; i <= to; ++i) {
-                elem = document.getElementById(i);
-                if (!elem) {
-                    break;
-                }
-                addClass(elem, "line-highlighted");
-            }
-        } else if (ev !== null && search && !hasClass(search, "hidden") && ev.newURL) {
+            return highlightSourceLines(match, ev);
```

Comment:
If you simply called this unconditionally, passing `undefined` for `match` then you wouldn't need to duplicate the match expression in this function and in `hilightSourceLines` -- also you'd be able to eliminate the duplication of the call to `hideSidebar()`

Reply:
You could have `highlightSourceLines()`  return whether it did work, and use that to decide if you want to show the search element.

Reply:
I'm not a big fan of the idea... :-/ If it matches, then we shouldn't look anywhere else. The code is more clear from my point of view in this way. Also, what would the boolean mean? If we encounter an error while trying to highlight the lines, should we return `true` or `false`? :)

Reply:
Okay that's fair, you can resolve this comment without change.